contract/src/test/kotlin/io/provenance/scope/loan/test/KotestHelpers.kt

```diff
@@ -124,10 +165,21 @@ private val anyTimestampComponents: Arb<Pair<Long, Int>> = Arb.pair(
 )
 
 private val anyPastNonEpochTimestampComponents: Arb<Pair<Long, Int>> = Instant.now().let { now ->
-    Arb.pair(
-        Arb.long(min = Timestamps.MIN_VALUE.seconds, max = now.epochSecond),
-        Arb.int(min = Timestamps.MIN_VALUE.nanos + 1, max = now.nano),
-    )
+    if (KotestConfig.runTestsExtended) {
+        Arb.pair(
+            Arb.long(min = Timestamps.MIN_VALUE.seconds, max = now.epochSecond),
+            Arb.int(min = Timestamps.MIN_VALUE.nanos, max = now.nano),
+        ).filterNot { (seconds, nanoSeconds) ->
+            seconds == Timestamps.MIN_VALUE.seconds && nanoSeconds == Timestamps.MIN_VALUE.nanos
+        }
+    } else {
+        Arb.pair(
+            Arb.long(min = Timestamps.MIN_VALUE.seconds, max = now.epochSecond),
+            Arb.int(min = Timestamps.MIN_VALUE.nanos + 1, max = now.nano), // Minor sacrifice of case where nanos = 0, in exchange for faster runs
+        )
+    }.filterNot { (seconds, nanoSeconds) ->
+        seconds == 0L && nanoSeconds == 0
+    }
```

Comment:
I'm sure this seems a bit strange at a glance, and I do think it is 😅 but usage of `filterNot` definitely can have a significant effect on runtime, especially in the case of this function where it feeds into the tedious `anyValidLoanState` logic. The idea of the `runTestsExtended` is to allow the creation of better Arbs without forcing polynomial runtime. I have yet to apply it everywhere though because sometimes it becomes a lengthy overkill for basic things (like the UUID Arbs).